src/Framework/PaymentGateways/PaymentGateway.php

```diff
@@ -128,13 +139,43 @@ public function createSubscription(GatewayPaymentData $paymentData, GatewaySubsc
 
 
     /**
-     * Handle gateway command
+     * Handle gateway route method
+     * @param  string  $method
+     * @param  array  $queryParams
      *
+     * @return void
      * @since 2.18.0
      *
+     * @unreleased - replace $donationId with $queryParams array
+     *
+     */
+    public function handleGatewayRouteMethod($method, $queryParams)
```

Comment:
I'm conflicted about having this within the gateway itself. Why doesn't this live within the framework?

Reply:
+1

Reply:
Honestly, I think these handle methods used to pull things in from the gateway in the earlier iterations - but now they're much more agnostic.  I'll abstract these out of the gateway class.